util/ranger/checker.go

```diff
@@ -24,93 +24,126 @@ import (
 
 // conditionChecker checks if this condition can be pushed to index planner.
 type conditionChecker struct {
-	checkerCol    *expression.Column
-	colUniqueID   int64
-	length        int
-	shouldReserve bool // check if a access condition should be reserved in filter conditions.
+	checkerCol               *expression.Column
+	length                   int
+	optPrefixIndexSingleScan bool
 }
 
-func (c *conditionChecker) check(condition expression.Expression) bool {
+func (c *conditionChecker) isFullLengthColumn() bool {
```

Comment:
I notice that this method is always used after `checkColumn`, and we almost always need to call this method after `checkColumn`.
Would it be better if we directly put this in `checkColumn`?

Reply:
I use a new function to wrap the logic as follows:
```
func (c *conditionChecker) matchColumn(expr expression.Expression) bool {
	// Check if virtual expression column matched
	if c.checkerCol != nil {
		return c.checkerCol.EqualByExprAndID(nil, expr)
	}
	return false
}

func (c *conditionChecker) checkColumn(expr expression.Expression) (isAccessCond, shouldReserve bool) {
	if c.matchColumn(expr) {
		return true, !c.isFullLengthColumn()
	}
	return false, true
}
```